File: backend/app/schemas/candidate.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, EmailStr, Field


class CandidateCreate(BaseModel):
    full_name: str = Field(..., min_length=2, max_length=255)
    email: EmailStr
    phone: str | None = None


class CandidateOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    full_name: str
    email: EmailStr
    phone: str | None
    created_at: datetime


class ResumeOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    candidate_id: str
    file_url: str
    parsed_skills: list[str]
    parsed_experience_years: float | None
    parsed_education: list[dict]
    parse_status: str
    created_at: datetime

File: frontend/types/candidate.ts
export interface Candidate {
  id: string;
  full_name: string;
  email: string;
  phone: string | null;
  created_at: string;
}

export interface CandidateCreatePayload {
  full_name: string;
  email: string;
  phone?: string;
}

export type ResumeParseStatus = "pending" | "parsed" | "failed";

export interface Resume {
  id: string;
  candidate_id: string;
  file_url: string;
  parsed_skills: string[];
  parsed_experience_years: number | null;
  parsed_education: Record<string, string>[];
  parse_status: ResumeParseStatus;
  created_at: string;
}
